File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/1616-minimum-difference-between-largest-and-smallest-value-in-three-moves.go
func minDifference(nums []int) int {
    if(len(nums) <= 4){
        return 0;
    }
    sort.Ints(nums);
    minDiff := int(^uint(0) >> 1)
   for left := 0 ; left <= 3 ;left++{
    diff := nums[len(nums) - 4 + left] - nums[left]
    if(diff < minDiff){
        minDiff = diff
    }
   }
   return minDiff
}

File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/1616-minimum-difference-between-largest-and-smallest-value-in-three-moves.java
class Solution {
    public int minDifference(int[] nums) {
        
        if(nums.length<=4){
            return 0;
        }
        Arrays.sort(nums);
        int minDiff = Integer.MAX_VALUE;
        for(int left =0 ;left <=3;left++ ){
            minDiff = Math.min(minDiff , nums[nums.length- 4 + left] - nums[left] );
        }
     return minDiff;
    }
}



















